src/main/java/org/eln2/mc/common/cell/AbstractCell.kt

```diff
@@ -0,0 +1,87 @@
+package org.eln2.mc.common.cell
+
+import net.minecraft.core.BlockPos
+import net.minecraft.resources.ResourceLocation
+import org.eln2.libelectric.sim.electrical.mna.component.Component
+import org.eln2.mc.common.In
+import org.eln2.mc.common.Side
+import org.eln2.mc.common.blocks.CellBlockEntity
+import org.eln2.mc.utility.DataLabelBuilder
+import org.eln2.mc.utility.McColor
+import org.eln2.mc.utility.SuffixConverter
+
+class ComponentInfo(val component: Component, val index : Int)
+
+abstract class AbstractCell(val pos : BlockPos) {
+    lateinit var id : ResourceLocation
+    lateinit var graph: CellGraph
+    lateinit var connections : ArrayList<AbstractCell>
+    var tile : CellBlockEntity? = null
+
+    /**
+     * Called when the tile entity is being unloaded.
+     * After this method is called, the field will be null.
+    */
+    open fun tileUnloaded(){}
+
+    /**
+     * Called when the tile entity is being loaded.
+     * The field is assigned before this is called.
+    */
+    open fun tileLoaded(){}
+
+    fun hasGraph() : Boolean { return this::graph.isInitialized }
+
+    /**
+     * Called when the graph manager completed loading this cell from the disk.
+    */
+    @In(Side.LogicalServer)
+    open fun completeDiskLoad(){}
+
+    /**
+     *   Called when the tile entity placing is complete.
+    */
+    @In(Side.LogicalServer)
+    open fun setPlaced(){}
+
+    /**
+     * Called when the tile entity is destroyed.
+    */
+    @In(Side.LogicalServer)
+    open fun destroy() {
+        graph.removeCell(this)
+    }
+
+    /**
+     * Called when the graph and/or neighbouring cells are updated. This method is called after completeDiskLoad and setPlaced
+     * @param connectionsChanged True if the neighbouring cells changed.
+     * @param graphChanged True if the graph that owns this cell has been updated.
+    */
+    @In(Side.LogicalServer)
+    open fun update(connectionsChanged : Boolean, graphChanged : Boolean){}
+
+    /**
+     * This method returns the text that will be displayed in the Circuit Explorer.
+    */
+    open fun createDataPrint() : DataLabelBuilder{
+        return DataLabelBuilder()
+    }
```

Comment:
This used to be a `String` and I kinda liked it better like that, but this also works.

My only thing is that we might support `IDetail` from `libage` here in the future so being able to return a `String` would be cool too.

Since this is only used for the plotter right now I'm not too attached but I feel like being able to dump the state of the graphs in a plain-text format would be really useful (perhaps we should just write a toString implementation?).